src/spikeinterface/postprocessing/correlograms.py

```diff
@@ -230,11 +267,40 @@ def compute_correlograms_numpy(sorting, window_size, bin_size):
 
 def correlogram_for_one_segment(spike_times, spike_labels, window_size, bin_size):
     """
-    Called by compute_correlograms_numpy
-    """
+    A very well optimized algorithm for the cross-correlation of
+    spike trains, copied from the Phy package, written by Cyrille Rossant.
+
+    For all spikes, time difference between this spike and
+    every other spike within the window is directly computed
+    and stored as a count in the relevant lag time bin.
+
+    Initially, the spike_times array is shifted by 1 position, and the difference
+    computed. This gives the time differences between the closest spikes
+    (skipping the zero-lag case). Next, the differences between
+    spikes times in samples are converted into units relative to
+    bin_size ('binarized'). Spikes in which the binarized difference to
+    their closest neighbouring spike is greater than half the bin-size are
+    masked.
+
+    Finally, the indices of the (num_units, num_units, num_bins) correlogram
+    that need incrementing are done so with `ravel_multi_index()`. This repeats
+    for all shifts along the spike_train until no spikes have a corresponding
+    match within the window size.
```

Comment:
So this would be another example where all this info could be here or go in Notes.